src/dcos_e2e/backends/_aws/__init__.py

```diff
@@ -0,0 +1,424 @@
+        agent_hosts = list(itertools.islice(cluster_hosts_iter, agents))
+        public_agent_hosts = list(
+            itertools.islice(cluster_hosts_iter, public_agents)
+        )
+
+        def nodes_from_hosts(hosts: Iterable[Host]) -> Set[Node]:
+            return set(
+                [
+                    Node(
+                        ip_address=IPv4Address(host.public_ip),
+                        ssh_key_path=ssh_key_path,
+                        private_ip_address=IPv4Address(host.private_ip),
+                    ) for host in hosts
+                ]
+            )
+
+        self._bootstrap_host = Node(
+            ip_address=IPv4Address(bootstrap_host.public_ip),
+            ssh_key_path=ssh_key_path,
+            private_ip_address=IPv4Address(bootstrap_host.private_ip),
+        )
+
+        self._masters = nodes_from_hosts(master_hosts)
+        self._agents = nodes_from_hosts(agent_hosts)
+        self._public_agents = nodes_from_hosts(public_agent_hosts)
+
+        self._default_ssh_user = self._aws_backend.default_ssh_user
+
+    def install_dcos_from_path(
+        self,
+        build_artifact: Path,
+        extra_config: Dict[str, Any],
+        log_output_live: bool,
+    ) -> None:
+        """
+        Args:
+            build_artifact: The `Path` to a build artifact to install DC/OS
+                from.
+            extra_config: May contain extra installation configuration
+                variables that are applied on top of the default DC/OS
+                configuration of the AWS backend.
+            log_output_live: If `True`, log output of the installation live.
+
+        Raises:
+            NotImplementedError: `NotImplementedError` because the AWS
+                backend can be utilized more efficiently by installing DC/OS
+                from from a URL using the advanced installation method.
+        """
+        message = (
+            'The AWS backend does not support the installing DC/OS from '
+            'a build artifact on the file system. This is due a more '
+            'efficient method existing in `install_dcos_from_url`.'
+        )
+        raise NotImplementedError(message)
+
+    def install_dcos_from_url(
+        self,
+        build_artifact: str,
+        extra_config: Dict[str, Any],
+        log_output_live: bool,
+    ) -> None:
+        """
+        Args:
+            build_artifact: The URL string to a build artifact to install DC/OS
+                from.
+            extra_config: This may contain extra installation configuration
+                variables that are applied on top of the default DC/OS
+                configuration of the AWS backend.
+            log_output_live: If `True`, log output of the installation live.
+        """
+        ssh_user = self._default_ssh_user
+
+        config = {
+            'cluster_name':
+            self._cf_stack.stack_id,
+            'resolvers': ['8.8.8.8'],
+            'dns_search':
```

Comment:
I'd really like a solid understanding of the differences between this and the equivalent in the Docker backend. If possible, I we should bring the creation of this configuration to the `Cluster` level and share it between backends.